drivers/gpu/drm/panel/msm8916-generated/panel-lenovo-phab-jdi_1080p.c

```diff
@@ -0,0 +1,456 @@
+		return 0;
+
+	ret = jdi_off(ctx);
+	if (ret < 0)
+		dev_err(dev, "Failed to un-initialize panel: %d\n", ret);
+
+	gpiod_set_value_cansleep(ctx->reset_gpio, 1);
+	regulator_bulk_disable(ARRAY_SIZE(ctx->supplies), ctx->supplies);
+
+	ctx->prepared = false;
+	return 0;
+}
+
+static const struct drm_display_mode jdi_mode = {
+	.clock = (720 + 72 + 16 + 60) * (1280 + 8 + 4 + 4) * 60 / 1000,
+	.hdisplay = 720,
+	.hsync_start = 720 + 72,
+	.hsync_end = 720 + 72 + 16,
+	.htotal = 720 + 72 + 16 + 60,
+	.vdisplay = 1280,
+	.vsync_start = 1280 + 8,
+	.vsync_end = 1280 + 8 + 4,
+	.vtotal = 1280 + 8 + 4 + 4,
+	.width_mm = 61,
+	.height_mm = 110,
+};
+
+static int jdi_get_modes(struct drm_panel *panel,
+			 struct drm_connector *connector)
+{
+	struct drm_display_mode *mode;
+
+	mode = drm_mode_duplicate(connector->dev, &jdi_mode);
+	if (!mode)
+		return -ENOMEM;
+
+	drm_mode_set_name(mode);
+
+	mode->type = DRM_MODE_TYPE_DRIVER | DRM_MODE_TYPE_PREFERRED;
+	connector->display_info.width_mm = mode->width_mm;
+	connector->display_info.height_mm = mode->height_mm;
+	drm_mode_probed_add(connector, mode);
+
+	return 1;
+}
+
+static const struct drm_panel_funcs jdi_panel_funcs = {
+	.prepare = jdi_prepare,
+	.unprepare = jdi_unprepare,
+	.get_modes = jdi_get_modes,
+};
+
+static int jdi_probe(struct mipi_dsi_device *dsi)
+{
+	struct device *dev = &dsi->dev;
+	struct jdi *ctx;
+	int ret;
+
+	ctx = devm_kzalloc(dev, sizeof(*ctx), GFP_KERNEL);
+	if (!ctx)
+		return -ENOMEM;
+
+	ctx->supplies[0].supply = "vsp";
+	ctx->supplies[1].supply = "vsn";
+	ret = devm_regulator_bulk_get(dev, ARRAY_SIZE(ctx->supplies),
+				      ctx->supplies);
+	if (ret < 0)
+		return dev_err_probe(dev, ret, "Failed to get regulators\n");
+
+	ctx->reset_gpio = devm_gpiod_get(dev, "reset", GPIOD_OUT_HIGH);
+	if (IS_ERR(ctx->reset_gpio))
+		return dev_err_probe(dev, PTR_ERR(ctx->reset_gpio),
+				     "Failed to get reset-gpios\n");
+
+	ctx->dsi = dsi;
+	mipi_dsi_set_drvdata(dsi, ctx);
+
+	dsi->lanes = 4;
+	dsi->format = MIPI_DSI_FMT_RGB888;
+	dsi->mode_flags = MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST |
+			  MIPI_DSI_MODE_NO_EOT_PACKET |
+			  MIPI_DSI_CLOCK_NON_CONTINUOUS | MIPI_DSI_MODE_LPM;
+
+	drm_panel_init(&ctx->panel, dev, &jdi_panel_funcs,
+		       DRM_MODE_CONNECTOR_DSI);
+
+	ret = drm_panel_of_backlight(&ctx->panel);
+	if (ret)
+		return dev_err_probe(dev, ret, "Failed to get backlight\n");
+
+	drm_panel_add(&ctx->panel);
+
+	ret = mipi_dsi_attach(dsi);
+	if (ret < 0) {
+		dev_err(dev, "Failed to attach to DSI host: %d\n", ret);
+		drm_panel_remove(&ctx->panel);
+		return ret;
+	}
+
+	return 0;
+}
+
+static void jdi_remove(struct mipi_dsi_device *dsi)
+{
+	struct jdi *ctx = mipi_dsi_get_drvdata(dsi);
+	int ret;
+
+	ret = mipi_dsi_detach(dsi);
+	if (ret < 0)
+		dev_err(&dsi->dev, "Failed to detach from DSI host: %d\n", ret);
+
+	drm_panel_remove(&ctx->panel);
+}
+
+static const struct of_device_id jdi_of_match[] = {
+	{ .compatible = "lenovo,phab-jdi_1080p" }, // FIXME
```

Comment:
```suggestion
	{ .compatible = "lenovo,phab-jdi-1080p" }, // FIXME
```

You also need to fix it in msm8916-mainline/linux-panel-drivers#28 and generate the driver again.

Reply:
can I use `phab-panel`?

Reply:
Only when you have defined `lenovo,phab-panel` in lk2nd and it selects from multiple panels.
https://github.com/msm8916-mainline/lk2nd/commit/933fbc3d178a101942869dea8b16a8651a1f7441

Anyway here is the naming rule, `<vendor>,[device-]<panel_name>`.

Reply:
> `<vendor>,[device-]<panel_name>`.

I see **panel_name**, not panel-name

Reply:
Basically it's meant to be dashes instead of underscores for compatible entries, if you know what I mean.